src/mlpack/tests/io_test.cpp

```diff
@@ -975,40 +999,41 @@ BOOST_AUTO_TEST_CASE(MatrixAndDatasetInfoTest)
   DatasetInfo info = move(get<0>(IO::GetParam<TupleType>("dataset")));
   arma::mat dataset = move(get<1>(IO::GetParam<TupleType>("dataset")));
 
-  BOOST_REQUIRE_EQUAL(info.Dimensionality(), 3);
+  REQUIRE(info.Dimensionality() == 3);
 
-  BOOST_REQUIRE(info.Type(0) == Datatype::categorical);
-  BOOST_REQUIRE_EQUAL(info.NumMappings(0), 3);
-  BOOST_REQUIRE(info.Type(1) == Datatype::numeric);
-  BOOST_REQUIRE(info.Type(2) == Datatype::categorical);
-  BOOST_REQUIRE_EQUAL(info.NumMappings(2), 2);
+  REQUIRE(info.Type(0) == Datatype::categorical);
+  REQUIRE(info.NumMappings(0) == 3);
+  REQUIRE(info.Type(1) == Datatype::numeric);
+  REQUIRE(info.Type(2) == Datatype::categorical);
+  REQUIRE(info.NumMappings(2) == 2);
 
-  BOOST_REQUIRE_EQUAL(dataset.n_rows, 3);
-  BOOST_REQUIRE_EQUAL(dataset.n_cols, 4);
+  REQUIRE(dataset.n_rows == 3);
+  REQUIRE(dataset.n_cols == 4);
 
   // The first dimension must all be different (except the ones that are the
   // same).
-  BOOST_REQUIRE_EQUAL(dataset(0, 0), dataset(0, 3));
-  BOOST_REQUIRE_NE(dataset(0, 0), dataset(0, 1));
-  BOOST_REQUIRE_NE(dataset(0, 1), dataset(0, 2));
-  BOOST_REQUIRE_NE(dataset(0, 2), dataset(0, 0));
+  REQUIRE(dataset(0, 0) == dataset(0, 3));
+  REQUIRE(dataset(0, 0) != dataset(0, 1));
+  REQUIRE(dataset(0, 1) != dataset(0, 2));
+  REQUIRE(dataset(0, 2) != dataset(0, 0));
 
-  BOOST_REQUIRE_CLOSE(dataset(1, 0), 1.0, 1e-5);
-  BOOST_REQUIRE_CLOSE(dataset(1, 1), 2.34, 1e-5);
-  BOOST_REQUIRE_CLOSE(dataset(1, 2), 1.03e5, 1e-5);
-  BOOST_REQUIRE_CLOSE(dataset(1, 3), -1.3, 1e-5);
+  REQUIRE(dataset(1, 0) == Approx(1.0).epsilon(1e-7));
+  REQUIRE(dataset(1, 1) == Approx(2.34).epsilon(1e-7));
+  REQUIRE(dataset(1, 2) == Approx(1.03e5).epsilon(1e-7));
+  REQUIRE(dataset(1, 3) == Approx(-1.3).epsilon(1e-7));
 
-  BOOST_REQUIRE_EQUAL(dataset(2, 0), dataset(2, 2));
-  BOOST_REQUIRE_EQUAL(dataset(2, 1), dataset(2, 3));
-  BOOST_REQUIRE_NE(dataset(2, 0), dataset(2, 1));
+  REQUIRE(dataset(2, 0) == dataset(2, 2));
+  REQUIRE(dataset(2, 1) == dataset(2, 3));
+  REQUIRE(dataset(2, 0) != dataset(2, 1));
 
   remove("test.arff");
 }
 
 /**
  * Test that we can access a parameter before we load it.
  */
-BOOST_AUTO_TEST_CASE(RawIntegralParameter)
+TEST_CASE_METHOD(IOTestDestroyer, "RawIntegralParameter",
+                "[IOTest]")
```

Comment:
```suggestion
TEST_CASE_METHOD(IOTestDestroyer, "RawIntegralParameter", "[IOTest]")
```

Looks like this would fit into the same line.